web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx

```diff
@@ -350,6 +371,111 @@ const CreateAccessRole = ({
   );
 };
 
+type TableData = {
+  items: SecurityGroup[];
+  nextToken?: string;
+  fetchStatus: FetchStatus;
+};
+
+const SelectSecurityGroups = ({
+  selectedSecurityGroups,
+  setSelectedSecurityGroups,
+  dbMeta,
+  emitErrorEvent,
+}: {
+  selectedSecurityGroups: string[];
+  setSelectedSecurityGroups(sg: string[]): void;
+  dbMeta: DbMeta;
+  emitErrorEvent(err: string): void;
+}) => {
+  const [sgTableData, setSgTableData] = useState<TableData>({
+    items: [],
+    nextToken: '',
+    fetchStatus: 'disabled',
+  });
+
+  const {
+    attempt: fetchSecurityGroupsAttempt,
+    setAttempt: setFetchSecurityGroupsAttempt,
+  } = useAttempt('');
+
+  function onSelectSecurityGroup(
+    sg: SecurityGroup,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) {
+    if (e.target.checked) {
+      return setSelectedSecurityGroups([...selectedSecurityGroups, sg.id]);
+    } else {
+      setSelectedSecurityGroups(
+        selectedSecurityGroups.filter(id => id !== sg.id)
+      );
+    }
+  }
+
+  async function fetchSecurityGroups() {
+    const integration = dbMeta.integration;
+    const selectedDb = dbMeta.selectedAwsRdsDb;
+
+    setFetchSecurityGroupsAttempt({ status: 'processing' });
+    try {
+      const { securityGroups, nextToken } =
+        await integrationService.fetchSecurityGroups(integration.name, {
+          vpcId: selectedDb.vpcId,
+          region: selectedDb.region,
+          nextToken: sgTableData.nextToken,
+        });
+
+      setFetchSecurityGroupsAttempt({ status: 'success' });
+      setSgTableData({
+        nextToken: nextToken,
+        fetchStatus: nextToken ? '' : 'disabled',
+        items: [...sgTableData.items, ...securityGroups],
+      });
+    } catch (err) {
+      const errMsg = getErrMessage(err);
+      setFetchSecurityGroupsAttempt({ status: 'failed', statusText: errMsg });
+      emitErrorEvent(`fetch security groups error: ${errMsg}`);
+    }
+  }
+
+  useEffect(() => {
+    fetchSecurityGroups();
+  }, []);
+
+  return (
+    <>
+      <Text bold>Step 3</Text>
+      <Text bold>Optionally Select Security Groups</Text>
+      <Text mb={2}>
+        Select security groups to assign to the Fargate service that will be
+        running the database access agent. The security groups you pick must
+        allow outbound connectivity to this Teleport cluster. If you don't
+        select any security groups, the default one for the VPC will be used.
+      </Text>
+      {fetchSecurityGroupsAttempt.status === 'failed' && (
+        <Danger>{fetchSecurityGroupsAttempt.statusText}</Danger>
+      )}
+      {fetchSecurityGroupsAttempt.status === 'processing' && (
+        <Flex width="904px" justifyContent="center" mt={3}>
+          <Indicator />
+        </Flex>
+      )}
```

Comment:
was it by design that security picker came after label matcher? or can security group come before label matcher? 

b/c the UX is also kinda weird to have a loader like this and a pressable (deploy) button right below it. if we can switch it up, the security group can load all it wants (or fail) and a user can just not care about it and skip to the next sub step with the deploy button

Reply:
I've changed the layout to make it a bit more clear and updated the screenshot in the PR description. I ran it by @roraback and got his approval.